gadget-container/entrypoint.sh

```diff
@@ -56,21 +59,53 @@ if grep -q '^1:name=systemd:.*/crio-[0-9a-f]*\.scope$' /proc/self/cgroup > /dev/
     CRIO=1
 fi
 
-if grep -q '^ID="rhcos"$' /host/etc/os-release > /dev/null ; then
-  if [ ! -d "/host/usr/src/kernels/$(uname -r)" ] ; then
-    echo "Fetching kernel-devel from CentOS 8."
-    REPO=http://mirror.centos.org/centos/8/BaseOS/x86_64/os/Packages/
-    RPM=kernel-devel-$(uname -r).rpm
-    RPMDIR=/opt/gadget-kernel/
-    RPMHOSTDIR=/host${RPMDIR}
-    mkdir -p $RPMHOSTDIR/usr/src/kernels/
-    test -r $RPMHOSTDIR/$RPM || \
-        curl -fsSLo $RPMHOSTDIR/$RPM $REPO/$RPM
-    test -r $RPMHOSTDIR/usr/src/kernels/`uname -r`/.config || \
-        chroot /host sh -c "cd $RPMDIR && rpm2cpio $RPM | cpio -i"
-    test ! -L /usr/src || rm -f /usr/src
-    mkdir -p /usr/src/kernels/`uname -r`/
-    mount --bind $RPMHOSTDIR/usr/src/kernels/`uname -r` /usr/src/kernels/`uname -r`
+if [ "$ID" = "rhcos" ] && [ ! -d "/host/usr/src/kernels/$KERNEL" ]; then
+  echo "Fetching kernel-devel from CentOS Vault Mirror..."
+
+  # CentOS version is composed by <RHEL_VERSION>.<RELEASE_YEAR><RELEASE_MONTH>
+  # https://access.redhat.com/articles/3078#RHEL8
+  case $RHEL_VERSION in
+    "8")
+      CENTOS_VERSION="8.0.1905"
+      ;;
+    "8.1")
+      CENTOS_VERSION="8.1.1911"
+      ;;
+    "8.2")
+      CENTOS_VERSION="8.2.2004"
+      ;;
+    "8.3")
+      CENTOS_VERSION="8.3.2011"
+      ;;
+    "8.4")
+      CENTOS_VERSION="8.4.2105"
+      ;;
+    "8.5")
+      CENTOS_VERSION="8.5.2111"
+      ;;
+    *)
+      echo "Couldn't retrieve CentOS repository from RHEL version $RHEL_VERSION" >&2
+      ;;
+  esac
+
+  if [ ! -z $CENTOS_VERSION ]; then
+    REPO=http://vault.centos.org/$CENTOS_VERSION/BaseOS/x86_64/os/Packages
+    RPMDIR=/tmp/gadget-kernel/$KERNEL
+    RPM=kernel-devel-$KERNEL.rpm
+
+    mkdir -p $RPMDIR/usr/src/kernels/
+    curl -fsSLo $RPMDIR/$RPM $REPO/$RPM || true
```

Comment:
Why `|| true`?

Reply:
I don't want the script to exit if this command fails but just print a warning [here](https://github.com/kinvolk/inspektor-gadget/blob/jose/centos-headers/gadget-container/entrypoint.sh#L105) given that Inspektor Gadget can anyway work using CORE gadgets. What do you think?

Reply:
Oh right, it makes sense.